File: 0739_Daily Temperatures/DailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<>();
        int[] ans = new int[temperatures.length];
        for(int i=temperatures.length - 1;i>=0;i--) {
            while(!stack.isEmpty() && temperatures[i] >= temperatures[stack.peek()]) stack.pop();
            ans[i] = stack.isEmpty() ? 0 : stack.peek() - i;
            stack.push(i);
        }
        
        return ans;
    }
}

File: 0739_Daily Temperatures/DailyTemperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        var n = temperatures.Length;
        var ans = new int[n];
        if(n == 0) return ans;
        
        var stack = new Stack<int>();
        
        for(int i=temperatures.Length-1;i>=0;i--)
        {
            while(stack.Count > 0 && temperatures[stack.Peek()] <= temperatures[i]) 
                stack.Pop();
            
            if(stack.Count > 0)
                ans[i] = stack.Peek() - i;  
            
             stack.Push(i);
        }
        
        return ans;
    }
}